packages/@react-spectrum/searchwithin/src/SearchWithin.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, SlotProvider, useFocusableRef, useResizeObserver, useStyleProps} from '@react-spectrum/utils';
+import {Field} from '@react-spectrum/label';
+import {FocusableRef} from '@react-types/shared';
+import React, {useCallback, useLayoutEffect, useRef, useState} from 'react';
+import {SpectrumSearchWithinProps} from '@react-types/searchwithin';
+import styles from '@adobe/spectrum-css-temp/components/searchwithin/vars.css';
+import {useLabel} from '@react-aria/label';
+import {useProvider, useProviderProps} from '@react-spectrum/provider';
+
+function SearchWithin(props: SpectrumSearchWithinProps, ref: FocusableRef<HTMLElement>) {
+  props = useProviderProps(props);
+  let {styleProps} = useStyleProps(props);
+  let {labelProps, fieldProps} = useLabel(props);
+  let {
+    children,
+    isDisabled,
+    isRequired,
+    label,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby
+  } = props;
+
+  let inputRef = useRef<HTMLInputElement | HTMLTextAreaElement>();
+  let domRef = useFocusableRef(ref, inputRef);
+
+  // Measure the width of the field to inform the width of the menu.
+  let [menuWidth, setMenuWidth] = useState(null);
+  let {scale} = useProvider();
+
+  let onResize = useCallback(() => {
+    if (domRef.current) {
+      let width = domRef.current.offsetWidth;
+      setMenuWidth(width);
+    }
+  }, [domRef]);
+
+  useResizeObserver({
+    ref: domRef,
+    onResize: onResize
+  });
+
+  useLayoutEffect(onResize, [scale, onResize]);
+
+  let defaultSlotValues = {isDisabled, isRequired, label: null, isQuiet: false, 'aria-labelledby': labelProps.id || ariaLabel};
```

Comment:
The `ariaLabel` shouldn't be a part of the `aria-labelledby`
```suggestion
  let defaultSlotValues = {isDisabled, isRequired, label: null, isQuiet: false, 'aria-labelledby': labelProps.id};
```

The tricky bit here is where should that `aria-label` go... I would assume it goes on the group itself, and perhaps the searchfield and picker have a aria-labelledby that points to the group? However, https://www.w3.org/TR/WCAG20-TECHS/ARIA17.html#:~:text=The%20group%20can%20be%20labeled,with%20role%3D%22group%22%20 only mentions `groups` being labelled by `aria-labelledby`. Digging to see if this is the right approach

cc @majornista

Reply:
@LFDanLu  `labelProps` will be empty if no `label` is provided. What `aria-labelledby` should the slots get in that case?

Reply:
I think `DatePicker` may be instructive on this point. 

- We use `useLabels` hook to retrieve the `aria-label` and `aria-labelledby` props: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/datepicker/src/useDatePicker.ts#L46-L47.
- `aria-label` for the `group` should probably default to `Search` if no `aria-label` or `aria-labelledby` prop is provided.
- The `id`, `aria-label` and `aria-labelledby` props returned by `useLabels` are applied to the `group`: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/datepicker/src/useDatePicker.ts#L89.
- The `aria-labelledby` prop returned by `useLabels`, which will reference either the `group` or external element(s) by IDREF, is then applied to the input fields: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/datepicker/src/useDatePicker.ts#L101.
- The dropdown menu button will then have a unique `id`, an `aria-label` like "Search within", and `aria-labelledby` concatenating the `aria-labelledby` prop returned by `useLabels` and the button `id`.

Reply:
Thanks @majornista . This will be addressed in a follow-up PR.